src/utill.rs

```diff
@@ -526,6 +526,30 @@ mod tests {
         );
     }
 
+    #[test]
```

Comment:
@Jem256 You can incorporate all these failure cases into a single test method.

Reply:
> @Jem256 You can incorporate all these failure cases into a single test method.

I have done that.